src/npcmove.cpp

```diff
@@ -756,7 +756,7 @@ npc_action npc::address_needs(int danger)
             return npc_undecided;
         }
 
-        if( has_effect( "allow_sleep" ) || fatigue > MASSIVE_FATIGUE ) {
+        if( misc_rules.allow_sleep || fatigue > MASSIVE_FATIGUE ) {
```

Comment:
So the effect is not used anymore? It's still preset in JSON, but removing it from there requires some code when loading the effects of a creature from a save, otherwise the game will complain about an invalid effect id.